src/media_tools/av_parsers.c

```diff
@@ -4385,6 +4501,461 @@ GF_Err gf_media_prores_parse_bs(GF_BitStream *bs, GF_ProResFrameInfo *prores_fra
+        *obu_size += (u32)gf_av1_leb128_read(bs, &leb128_size);
+        *obu_size += leb128_size;
+
+	u64 read_num_samples_to_trim_at_start = 0;
+	u64 read_num_samples_to_trim_at_end = 0;
+	if (obu_trimming_status_flag) {
+		read_num_samples_to_trim_at_end = gf_av1_leb128_read(bs, NULL);
+		read_num_samples_to_trim_at_start = gf_av1_leb128_read(bs, NULL);
+	}
+	if (num_samples_to_trim_at_start) {
+		*num_samples_to_trim_at_start = read_num_samples_to_trim_at_start;
+	}
+	if (num_samples_to_trim_at_end) {
+		*num_samples_to_trim_at_end = read_num_samples_to_trim_at_end;
+	}
+
+	if (obu_extension_flag) {
+		extension_header_size = (u32)gf_av1_leb128_read(bs, NULL);
+		for (i = 0; i < extension_header_size; ++i) {
+			/*extension_header_bytes=*/gf_bs_read_u8(bs);
+		}
+	}
+
+        return GF_OK;
+}
+
+static Bool iamf_is_profile_supported(u8 profile)
+{
+	return profile == 0 || profile == 1 || profile == 2;
+}
+
+static GF_Err iamf_parse_ia_sequence_header(GF_BitStream *bs)
+{
+	u32 ia_code = gf_bs_read_int_log(bs, 32, "ia_code");
+	if (ia_code != GF_4CC('i', 'a', 'm', 'f'))
+	{
+		return GF_NON_COMPLIANT_BITSTREAM;
+	}
+	u8 primary_profile = gf_bs_read_int_log(bs, 8, "primary_profile");
+	u8 additional_profile = gf_bs_read_int_log(bs, 8, "additional_profile");
+	if (iamf_is_profile_supported(primary_profile) || iamf_is_profile_supported(additional_profile))
+	{
+		return GF_OK;
+	}
+
+	// Maybe this is a future version of IAMF, which claims some backwards incompatibility.
+	return GF_NOT_SUPPORTED;
+}
+
+static GF_Err iamf_parse_codec_config(GF_BitStream *bs, IAMFState *state)
+{
+	GF_Descriptor *desc = NULL;
+	GF_Err e = GF_OK;
+	gf_av1_leb128_read(bs, NULL); // `codec_config_id`.
+	u32 codec_id = gf_bs_read_int_log(bs, 32, "codec_id");
+	state->num_samples_per_frame = gf_av1_leb128_read(bs, NULL);
+	state->audio_roll_distance = gf_bs_read_int_log(bs, 16, "roll_distance");
+	switch (codec_id) {
+	case GF_4CC('O', 'p', 'u', 's'):
+		state->sample_rate = 48000;
+		state->sample_size = 16;
+		break;
+	case GF_4CC('f', 'L', 'a', 'C'):
+		gf_bs_read_int_log(bs, 1, "last_metadata_block_flag");
+		gf_bs_read_int_log(bs, 7, "block_type");
+		gf_bs_read_int_log(bs, 24, "block_length");
+		gf_bs_read_int_log(bs, 16, "minimum_block_size");
+		gf_bs_read_int_log(bs, 16, "maximum_block_size");
+		gf_bs_read_int_log(bs, 24, "minimum_frame_size");
+		gf_bs_read_int_log(bs, 24, "maximum_frame_size");
+		state->sample_rate = gf_bs_read_int_log(bs, 24, "sample_rate");
+		gf_bs_read_int_log(bs, 3, "num_of_channels");
+		state->sample_size = gf_bs_read_int_log(bs, 5, "bits_per_sample");
+		break;
+	// LPCM.
+	case GF_4CC('i', 'p', 'c', 'm'):
+		gf_bs_read_int_log(bs, 8, "sample_format_flags");
+		state->sample_size = gf_bs_read_int_log(bs, 8, "sample_size");
+		state->sample_rate = gf_bs_read_int_log(bs, 32, "sample_rate");
+		break;
+	// AAC-LC.
+	case GF_4CC('m', 'p', '4', 'a'):
+		// Read `DecoderConfig`.
+		u32 unused_size;
```

Comment:
in C99 (which is the standard gpac uses), although you no longer have to declare all variables at the top of the function, you can't have a variable declaration immediately after a goto/case label, because declarations are not considered as statements, and you can only label statements

there's an interesting discussion about it here: https://stackoverflow.com/a/19830820

fix is either to scope the case in a `{}` block, or add a `;` after the label as an empty statement

Reply:
Done, thanks for the link.